lua/cokeline/buffers.lua

```diff
@@ -136,17 +136,23 @@ end
 
 ---@param path  string
 ---@param filename  string
----@param type  string
+---@param buftype  string
+---@param filetype  string
 ---@return table<string, string>
-local get_devicon = function(path, filename, type)
-  local name = (type == "terminal") and "terminal" or filename
-  local extn = fn.fnamemodify(path, ":e")
-  local icon, color =
-    rq_devicons.get_icon_color(name, extn, { default = true })
-  return {
-    icon = icon .. " ",
-    color = color,
-  }
+local get_devicon = function(path, filename, buftype, filetype)
+  local icon, color
+  if _G.cokeline.config.buffers.icon_by == "type" then
+    local type = (buftype == "terminal") and "terminal" or filetype
+    icon, color = rq_devicons.get_icon_color_by_filetype(type, { default = true })
```

Comment:
This is good, but maybe it should be used as a fallback instead?

Something like

```lua
local icon, color = get_icon_color(name, ext)
if not icon then
  icon, color = ...by_filetype(vim.bo.filetype, { default = true })
end
```

This way we get the advantage of more consistent icon fetching, but without any extra configuration required.

Reply:
This is a good idea !